src/models/ObjectLevelCan.ts

```diff
@@ -1,17 +1,21 @@
-type BasicPermissions = 'create' | 'read' | 'update' | 'delete'
-
-
 type ObjectLevelCan<TPermissions extends string> = {
   [key in TPermissions]: boolean
 }
 
-export type BasicPermissionsObjectLevelCan = ObjectLevelCan<BasicPermissions>
-export type BasicRunnablePermissionsObjectLevelCan = ObjectLevelCan<BasicPermissions | 'run'>
+export const BasicObjectLevelPermissions = ['create', 'read', 'update', 'delete'] as const
+export const RunnableObjectLevelPermissions = [...BasicObjectLevelPermissions, 'run'] as const
+
+export type BasicPermissionsObjectLevelCan = ObjectLevelCan<typeof BasicObjectLevelPermissions[number]>
+export type BasicRunnablePermissionsObjectLevelCan = ObjectLevelCan<typeof RunnableObjectLevelPermissions[number]>
 
-export function createObjectLevelCan<T extends Record<string, boolean>>(): T {
+export function createObjectLevelCan<T extends Record<string, boolean>>(knownProperties: typeof BasicObjectLevelPermissions | typeof RunnableObjectLevelPermissions): T {
```

Comment:
Thought:

Do we really care if only the exact permissions that object has return true as long as the types are correct? I'd vote not having this accept any arguments and just checking the array of known values directly. Especially since you're defining the known values here and exporting them just so the implementations can pass them back. Probably not worth the added complexity.

Reply:
Yeah i def. thought this - https://github.com/PrefectHQ/prefect-ui-library/pull/1810/files#r1355341844

figured i'd just lean toward explicit over implicit 🤷